### ISSUE TITLE
change actioncheck purpose

### ISSUE THREAD

**Issue by grooverdan** [fail2ban/fail2ban#488]
How about instead of the rather excessive checks of actioncheck (which is undoubtly almost always true) before each ban, the actioncheck gets executed if actionban fails. If the actioncheck also fails then a actionstop, actionstart can then put the action back in the right state.

(for 0.9.x)

**Comment by sebres** [comment]
The part with "check on error" was implemented in #2588 (and release >= 0.10.5).
So it's very easy to disable `actioncheck` in normal case on ban/unban (in sane environment) by controlling epoch (and/or some new flag parameter, like `check=False`).

**Comment by sebres** [comment]
This is fully implemented in ed4b5b8bdc8914e9d8dc68ca889f7c05ea194986 now,
although the enhancement looks safe, but I can not (and don't really want) check all the actions we have, whether they are safe against this change.
So unless it is known, v.1.0 will be the first version this "feature" will appear (firstly).

For the actions that would still need old handling (so check on ban or unban), it would be enough to do a change similar this:
```diff
- actionban = ban <ip>
+ actionban = %(actioncheck)s && ban <ip>
```
or possibly (without modifying the action) simply prefixed in jail/default (jail.local):
```ini
action = some-action[actionban='<actioncheck> && <known/actionban>']
```
but maybe we could extend it with some parameter that would made it easier.

**Comment by sebres** [comment]
Oh, I forgot to add: this seems to increase banning performance of single IP (up to 2x) for several actions.
